Source/WTF/wtf/ContinuousArenaMalloc.cpp

```diff
@@ -294,12 +308,12 @@ bool ContinuousArenaMalloc::extentPurgeLazy(extent_hooks_t *extent_hooks,
               length,
               arena_ind);
 
-    ASSERT(isValidRange(addr, size));
-    ASSERT(length <= size);
-
-    char *start = (char *)addr + offset;
+    ASSERT(isAllocatedRange(addr, size));
 
-    ASSERT(isValidRange(start, length));
+    char *start = reinterpret_cast<char *>(addr) + offset;
+    ASSERT(start >= reinterpret_cast<char *>(addr));
+    ASSERT((start + length) <= (reinterpret_cast<char *>(addr) + size));
```

Comment:
Do we need to check that `addr + size` won't overflow?

Reply:
That's a more complicated question that it appears. It would a misuse of the API, but that's true for all of these assertions. I would also argue that the earlier `isAllocatedRange` would check this, but actually it makes the same mistake. Also, the `(addr) + offset` overflow check is not really reliable because pointer overflow is undefined behaviour.

I'll consider using `uintptr_t` for these checks, since that does have well-defined overflow.

Reply:
I think this is resolved by [63d9255](https://github.com/capablevms/webkit/pull/5/commits/63d9255cce241ceee6f197b4b8d096f0732b406d) and [c990349](https://github.com/capablevms/webkit/pull/5/commits/c990349662a02accf9f11ded8410bfd627f7cbfa).